### ISSUE TITLE
Merge PMArbitraryPrecisionFloatForTrigonometry with PMArbitraryPrecisionFloat

### ISSUE THREAD

**Issue by SergeStinckwich** [PolyMathOrg/PolyMath#197]
@nicolas-cellier-aka-nice why do we need a separate class only for sin, cos and pi for arbitrary precision float``PMArbitraryPrecisionFloatForTrigonometry``? I think we can merge with ``PMArbitraryPrecisionFloat`` to avoid confusion.

**Comment by nicolas-cellier-aka-nice** [comment]
Yes, good idea, the only thing to care of is the precision of pi used in the subclass.

**Comment by SergeStinckwich** [comment]
Do you mind to do a PR with this modification? Thank you.

**Comment by nicolas-cellier-aka-nice** [comment]
I've implemented upstream, see issue #199

**Comment by SergeStinckwich** [comment]
ok thank you. I have to see how to see the changes with the upstream and the PM version.
